### ISSUE TITLE
key is removed when database gets locked because of duplicate key in recycle bin entry

### ISSUE THREAD

**Issue by CWempe** [dlech/KeeAgent#172]
Hello,

I want my keys to be loaded while the databse is locked.

I set the corresponding options.
![image](https://cloud.githubusercontent.com/assets/1270359/19526544/01dc6762-9625-11e6-95e3-7efaec1da270.png)
![image](https://cloud.githubusercontent.com/assets/1270359/19526608/323bcdb2-9625-11e6-879b-58eda154e5f7.png)

I have 3 keys in my database and they are all working when the database is unlocked.

But as soon as the database gets locked, one of my keys is removed from the KeeAagent.
It is always the key "putty@server".

I double checked and tripple checked the settings, but could not find any difference to the working key-entries.

I changed the comment and filename incase the "@" character would be the culprit.
I created new entries with the same key.
No success.

I created a new databse with a new generated key and this key.
It worked!

Does anybody have an idea, why this key is having problems with this database?
Are there any log files where I can check why the key is removed?

My System:
Win10 x64
KeePass 2.34
KeeAgent 0.8.1.0

**Comment by dlech** [comment]
The only thing I can think of is that the `KeeAgent.settings` for that entry got corrupted somehow.

![database kdbx - keepass_009](https://cloud.githubusercontent.com/assets/963645/19539696/6acc5b92-9620-11e6-9a73-55a1503f3962.png)

Can you paste the contents of that attachment for the problematic entry?

Deleting the attachment for that entry and setting it up again will probably fix it.

**Comment by CWempe** [comment]
I already tried that.

```
<?xml version="1.0" encoding="utf-16"?>
<EntrySettings xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <AllowUseOfSshKey>true</AllowUseOfSshKey>
  <AddAtDatabaseOpen>true</AddAtDatabaseOpen>
  <RemoveAtDatabaseClose>false</RemoveAtDatabaseClose>
  <UseConfirmConstraintWhenAdding>false</UseConfirmConstraintWhenAdding>
  <UseLifetimeConstraintWhenAdding>false</UseLifetimeConstraintWhenAdding>
  <LifetimeConstraintDuration>600</LifetimeConstraintDuration>
  <Location>
    <SelectedType>attachment</SelectedType>
    <AttachmentName>putty@server.ppk</AttachmentName>
    <SaveAttachmentToTempFile>false</SaveAttachmentToTempFile>
    <FileName />
  </Location>
</EntrySettings>
```

I will experiment further then.

**Comment by CWempe** [comment]
I solved my problem with a workaround.

I created a new database with the same settings.
Copied the key via clipboard from the old to the new database.
Tried successfully using the key while the databse was locked.
And finally imported all other entries from the old database file.

btw: Importing all entries (including the key) to a new database did not work. :/

Maybe this will help somebody in case anybody has the same problem.
You can close this issue if you like.
I don't think I can give you any more information without sending you my database. ;)

**Comment by dlech** [comment]
I wonder if you have an entry in the recycle bin with the same key. Or just two entries with the same key somehow.

**Comment by CWempe** [comment]
You are right!!!!
I just checked.

There was actually a second entry with the same key in my recycle bin.
And the "Remove key from agent when databse is closed/locked"-option was activated. :(

Thanks.

**Comment by dlech** [comment]
KeeAgent is supposed to ignore entries in the recycle bin, so I will try to see if I can reproduce and fix this.

**Comment by dlech** [comment]
Fix has been released in v0.9.1. http://lechnology.com/2017/03/keeagent-beta-0-9-1-released/
